### PR TITLE
169 Apply custom font and scale it dynamically

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

**Comment by zamorastp** [comment]
Simulator Screen Shot - iPhone 13 - 2022-06-06 at 12 32 37 copy
Simulator Screen Shot - iPhone 13 - 2022-06-06 at 12 32 33 copy
Simulator Screen Shot - iPhone 13 - 2022-06-06 at 12 32 28 copy
Simulator Screen Shot - iPhone 13 - 2022-06-06 at 12 32 24 copy

File: Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+// Copyright © 2022 Schibsted.
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import UIKit
+
+extension UIFont {
+    static func preferedCustomFont(_ fontFamily: String, textStyle: UIFont.TextStyle) -> UIFont {
```
Comment (oscarsilver) [inline Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift:9]: Nit: Preferred

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/FontManager.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Inter.allCases.forEach {
+            registerFont(bundle: Bundle.accountSDK(for: SimplifiedLoginViewController.self), fontName: $0.rawValue, fontExtension: "ttf")
+        }
+    }
+
+    fileprivate static func registerFont(bundle: Bundle, fontName: String, fontExtension: String) {
+        guard let fontURL = bundle.url(forResource: fontName, withExtension: fontExtension),
+              let fontDataProvider = CGDataProvider(url: fontURL as CFURL),
+              let font = CGFont(fontDataProvider) else {
+            fatalError("Couldn't create font from filename: \(fontName) with extension \(fontExtension)")
```
Comment (oscarsilver) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/FontManager.swift:22]: Do we really want to crash the app if we can't parse the font somehow?

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift [hunk 3]
```diff
@@ -25,7 +25,7 @@ final class ContinueButton: UIButton {
         let view = UILabel()
         view.text = viewModel.initials
         view.isAccessibilityElement = false
-        view.font = UIFont.boldSystemFont(ofSize: 17)
+        view.font = UIFont(name: Inter.bold.rawValue, size: 17)
```
Comment (oscarsilver) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift:28]: Suggestion: Maybe make an extension of the `UIFont` initializer that takes a case of the Inter enum? Or make a use of a protocol that Inter and other fonts can conform to and use that in an initializer? That way we don't have to write `.rawValue` everywhere

File: Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift (hunk 1 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift:9]: done

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/FontManager.swift (hunk 2 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/FontManager.swift:22]: If app crashes that it means that files in the sdk are somehow corrupted 🤔

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift (hunk 3 above)
Comment (clausjoergensen) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift:28]: The "Inter" font is only being used by the Norwegian news apps today. Why are we using it in a SDK component shared by many different Schibsted apps?

It'll look out of place in Aftonbladet, SvD, Bilbasen, etc.

Imo. a component like the simplified login should just stick to using San Fransisco.

File: Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import UIKit
+
+extension UIFont {
+    static func preferredCustomFont(_ fontFamily: Inter, textStyle: UIFont.TextStyle) -> UIFont {
+        let systemDescriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: textStyle)
+
+        let customFontDescriptor = UIFontDescriptor.init(
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Lib/Extensions/UIFont+Custom.swift:12]: Please don't use `.init`, just call the constructor normally:

```suggestion
        let customFontDescriptor = UIFontDescriptor(
```

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/Inter.swift [hunk 5]
```diff
@@ -0,0 +1,13 @@
+//
+// Copyright © 2022 Schibsted.
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import Foundation
+
+enum Inter: String, CaseIterable {
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/Fonts/Inter.swift:8]: Let's add a note that it's a open source font:

```suggestion
// Inter is a Open Font, see https://github.com/rsms/inter
enum Inter: String, CaseIterable {
```

**Comment by zamorastp** [comment]
Previous screenshots weren't correct. I attached the good ones now. @clausjoergensen @oscarsilver 

Simulator Screen Shot - iPhone 13 - 2022-06-08 at 15 18 39
Simulator Screen Shot - iPhone 13 - 2022-06-08 at 15 18 46
Simulator Screen Shot - iPhone 13 - 2022-06-08 at 15 18 51
Simulator Screen Shot - iPhone 13 - 2022-06-08 at 15 18 59

File: Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift (hunk 3 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift:28]: Ok, I have an explanation. The Inter font is the standard Schibsted font across the brands. It aligns with the strategy to make the Schibsted account(company) recognizable by the brand's users. The same font is used for account, privacy, and payment gateway services. This is the valid Schibsted font.
The new implementation of cookie consent banner from Bilbasen also uses Inter font. You can see it [here ](https://www.figma.com/file/1AxYNn0zEck0aEGcaHkqYi/Simplified-login-%E2%80%A2-Native-iOS-SDK%2FDesktop?node-id=1092%3A22995)

@clausjoergensen @oscarsilver

Comment (clausjoergensen) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/Sub Views/ContinueButton.swift:28]: Alright, that's a acceptable reason ;)

**Comment by clausjoergensen** [comment]
Tested in Sportbladet, seems fine

**Review by clausjoergensen — APPROVED** [review]

**Pull request merged**